test/box-luatest/gh_6310_grant_rw_access_on__session_settings_space_to_public_role_test.lua

```diff
@@ -0,0 +1,57 @@
+local cluster = require('test.luatest_helpers.cluster')
+local fio = require('fio')
+local t = require('luatest')
+
+local g = t.group('gh-6310-grant-rw-access-on-_session_settings-space-to-public-role')
+
+g.before_all(function()
+    local helpers = require('test.luatest_helpers')
+
+    g.cluster = cluster:new({})
+
+    local bootstrap_box_cfg = {
+        listen = helpers.instance_uri('bootstrap'),
+    }
+    g.bootstrap = g.cluster:build_and_add_server({alias   = 'bootstrap',
+                                                  box_cfg = bootstrap_box_cfg})
+
+    local upgrade_box_cfg = {
+        listen   = helpers.instance_uri('upgrade'),
+        work_dir = fio.pathjoin(fio.cwd(), 'test/box-luatest/upgrade/2.9.1')
+    }
+    g.upgrade = g.cluster:build_and_add_server({alias   = 'upgrade',
+                                                box_cfg = upgrade_box_cfg})
+
+    g.cluster:start()
+end)
+
+g.after_all(function()
+    g.cluster:drop()
+end)
+
+local msg_fmt = 'public role has read,write access on ' ..
+                '_session_settings space on %s instance'
+g.test_boostrap = function()
+    local privs = g.bootstrap:exec(function()
+        local public_privs = box.schema.role.info('public')
+        for _, priv in pairs(public_privs) do
+            if priv[3] == '_session_settings' then
+                return priv[1]
+            end
+        end
+    end)
+    t.assert(privs and privs:find('read,write'), msg_fmt:format('bootstrapped'))
+end
+
+g.test_upgrade = function()
+    local privs = g.upgrade:exec(function()
+        box.schema.upgrade()
+        local public_privs = box.schema.role.info('public')
+        for _, priv in pairs(public_privs) do
+            if priv[3] == '_session_settings' then
```

Comment:
I'd also add original test from the issue to make sure that privileges work fine:)